// mailtrap-client/dist/api/CompaniesEndpoint.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CompaniesEndpoint = void 0;
class CompaniesEndpoint {
    constructor(http) {
        this.http = http;
    }
    getCompanies() {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.request('GET', '/companies');
        });
    }
    getCompany(idCompany) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.request('GET', `/companies/${idCompany}`);
        });
    }
    patchCompany(idCompany, data) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.request('PATCH', `/companies/${idCompany}`, data);
        });
    }
    deleteCompany(idCompany) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.request('DELETE', `/companies/${idCompany}`);
        });
    }
    createCompanyInbox(idCompany, data) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.request('POST', `/companies/${idCompany}/inboxes`, data);
        });
    }
    getSharedUsers(idCompany) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.request('GET', `/companies/${idCompany}/shared_users`);
        });
    }
    getSharedUser(idCompany, idSharedUser) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.request('GET', `/companies/${idCompany}/shared_users/${idSharedUser}`);
        });
    }
    deleteSharedUser(idCompany, idSharedUser) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.request('DELETE', `/companies/${idCompany}/shared_users/${idSharedUser}`);
        });
    }
}
exports.CompaniesEndpoint = CompaniesEndpoint;
