// mailtrap-client/dist/api/CorsDomainsEndpoint.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CorsDomainsEndpoint = void 0;
class CorsDomainsEndpoint {
    constructor(http) {
        this.http = http;
    }
    getCorsDomains() {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.request('GET', `/cors_domains`);
        });
    }
    getCorsDomain(idCorsDomain) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.request('GET', `/cors_domains/${idCorsDomain}`);
        });
    }
    createCorsDomain(idCorsDomain) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.request('POST', `/cors_domains/${idCorsDomain}`);
        });
    }
    deleteCorsDomain(idCorsDomain) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.request('DELETE', `/cors_domains/${idCorsDomain}`);
        });
    }
    patchCorsDomain(idCorsDomain, data) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.request('PATCH', `/cors_domains/${idCorsDomain}`, data);
        });
    }
}
exports.CorsDomainsEndpoint = CorsDomainsEndpoint;
