// mailtrap-client/dist/MailtrapClient.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MailtrapClient = void 0;
const MessagesEndpoint_1 = require("./api/MessagesEndpoint");
const HttpClient_1 = require("./http/HttpClient");
const UserEndpoint_1 = require("./api/UserEndpoint");
const CompaniesEndpoint_1 = require("./api/CompaniesEndpoint");
const SharedInboxesEndpoint_1 = require("./api/SharedInboxesEndpoint");
const InboxesEndpoint_1 = require("./api/InboxesEndpoint");
const CorsDomainsEndpoint_1 = require("./api/CorsDomainsEndpoint");
const FetchHttpProvider_1 = require("./http/FetchHttpProvider");
class MailtrapClient {
    constructor(apiKey, provider = FetchHttpProvider_1.FetchProvider) {
        this.apiKey = apiKey;
        const urlBase = 'https://mailtrap.io/api/v1';
        const http = new HttpClient_1.HttpClient(this.apiKey, urlBase, provider);
        this.user = new UserEndpoint_1.UserEndpoint(http);
        this.companies = new CompaniesEndpoint_1.CompaniesEndpoint(http);
        this.sharedInboxes = new SharedInboxesEndpoint_1.SharedInboxesEndpoint(http);
        this.inboxes = new InboxesEndpoint_1.InboxesEndpoint(http);
        this.messages = new MessagesEndpoint_1.MessagesEndpoint(http);
        this.corsDomains = new CorsDomainsEndpoint_1.CorsDomainsEndpoint(http);
    }
}
exports.MailtrapClient = MailtrapClient;
